Dependency updates

michaelglenister opened the pull request:
Closes #520 #522 #527 #531 #631 #633 #659 #668

paulmwatson commented on requirements.txt:
```diff
@@ -31,17 +31,17 @@ gevent==1.4.0
 gunicorn==19.9.0
 idna==2.7
-ipython==4.1.2
+ipython==7.34.0
```
Are all of these used by the application code? Some like ipython might be from developer machines and not used?

michaelglenister commented:
Removed unused packages and fixed merge conflict however this pull request does not reflect the changes from its respective branch, haven't seen that before

michaelglenister commented:
Okay now the changes show up maybe caching error on my side, no idea
